app/engine/WorkoutRecorder.js

```diff
@@ -53,6 +55,41 @@ function createWorkoutRecorder () {
     await createFile(rotationImpulses.join('\n'), filename, config.gzipRawDataFiles)
   }
 
+  async function createRowingDataFile () {
+    const stringifiedStartTime = startTime.toISOString().replace(/T/, '_').replace(/:/g, '-').replace(/\..+/, '')
+    const directory = `${config.dataDirectory}/recordings/${startTime.getFullYear()}/${(startTime.getMonth() + 1).toString().padStart(2, '0')}`
+    const filename = `${directory}/${stringifiedStartTime}_rowingData.csv`
+    let currentstroke
+    let trackPointTime
+    let timestamp
+    let i
+
+    log.info(`saving session as RowingData file ${filename}...`)
+
+    // Required file header, please note this includes a typo and odd spaces as the specification demands it!
+    let RowingData = ',index, Stroke Number,TimeStamp (sec), ElapsedTime (sec), HRCur (bpm),DistanceMeters, Cadence (stokes/min), Stroke500mPace (sec/500m), Power (watts), StrokeDistance (meters),' +
+    ' DriveTime (ms), DriveLength (meters), StrokeRecoveryTime (ms),Speed, Horizontal (meters), Calories (kCal), DragFactor, PeakDriveForce (N), AverageDriveForce (N),' +
+    'Handle_Force_(N),Handle_Velocity_(m/s),Handle_Power_(W)\n'
+
+    // Add the strokes
+    i = 0
+    while (i < strokes.length) {
+      currentstroke = strokes[i]
+      trackPointTime = new Date(startTime.getTime() + currentstroke.totalMovingTime * 1000)
+      timestamp = trackPointTime / 1000
```

Comment:
trackPoint is of type Date, the left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type. JS allows implicit conversion but I recommend avoiding it. getTime() should be called before division

Reply:
This is more @laberning 's strong point than mine. @laberning ?